code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -630,19 +627,24 @@
 	user.verbs += /mob/living/proc/guardian_recall
 	user.verbs += /mob/living/proc/guardian_reset
 
-	var/picked_name = pick("Aries", "Leo", "Sagittarius", "Taurus", "Virgo", "Capricorn", "Gemini", "Libra", "Aquarius", "Cancer", "Scorpio", "Pisces")
-	var/colour = pick("orange", "pink", "red", "blue", "green")
-	G.name = "[picked_name] [capitalize(colour)]"
-	G.real_name = "[picked_name] [capitalize(colour)]"
-	G.icon_living = "guardian[colour]"
-	G.icon_state = "guardian[colour]"
-	G.icon_dead = "guardian[colour]"
-
-	switch (theme)
+	var/colour
+	var/picked_name
+	switch(theme)
 		if("magic")
 			user << "[G.magic_fluff_string]."
+			colour = pick("orange", "pink", "red", "blue", "green")
+			picked_name = pick("Aries", "Leo", "Sagittarius", "Taurus", "Virgo", "Capricorn", "Gemini", "Libra", "Aquarius", "Cancer", "Scorpio", "Pisces")
 		if("tech")
 			user << "[G.tech_fluff_string]."
+			colour = pick("lily", "orchid", "rose", "iris", "ivy", "zinnia")
```

Comment:
If you've got a more commonly known flower that can be greenish-yellow and that's one word feel free to tell me.